### DISCUSSION TITLE
part3.쿠버네티스 입문 - 05.[실습] kubectl 명렁어 맛보기 질문 드립니다.

### DISCUSSION THREAD

**Discussion by sc303030** [yoonjeong-kwon/fastcampus-kubernetes#23] [Q&A] [answered]
마지막 단계인 로그 확인 과정에서 실행되는 pods에 관하여 질문이 있습니다.
```shell
# 현재 실행중인 컨테이너 프로세스에 접속하여 로그 확인
kubectl attach deployment/nginx-deployment -c nginx

# 현재 실행중인 컨테이너 프로세스에 모든 로그 출력 (-f: watch 모드)
kubectl logs deployment/nginx-deployment -c nginx -f
```
위 명령어를 실행하기 위해 포트와 연결할 pods를 아래에서 맨 위에 있는 `nginx-deployment-64779c7579-7f7st`로 연결하였습니다.(포트포워딩)
```shell
PS C:\Users\gh\fastcampus-kubernetes-main\chapter01> kubectl get pods
NAME                                READY   STATUS    RESTARTS   AGE
nginx-deployment-64779c7579-7f7st   1/1     Running   0          20m
nginx-deployment-64779c7579-9xmvr   1/1     Running   0          21m
nginx-deployment-64779c7579-cc9qp   1/1     Running   0          18m
nginx-deployment-64779c7579-hb2pk   1/1     Running   0          21m
```
그러나 `kubectl attach deployment/nginx-deployment -c nginx`명령어를 입력하면 다른 pods를 사용하라고 나오더라구요.
```
PS C:\Users\gh\fastcampus-kubernetes-main\chapter01> kubectl logs deployment/nginx-deployment -c nginx -f
Found 4 pods, using pod/nginx-deployment-64779c7579-9xmvr
```
그래서인지 `kubectl logs deployment/nginx-deployment -c nginx -f`을 해도 로그가 찍히지 않았습니다.(`64779c7579-7f7st`로 포트 포워딩중)

이럴경우 먼저 attach를 이용하여 사용해야 할 pods를 확인해야 하나요?? 아니면 제가 놓쳤던 부분이 있던것인지 궁금합니다..!

**Comment by yoonjeong-kwon** [comment, marked as answer]
@sc303030 님, 안녕하세요^^

본문에 명령어 실행 결과를 적어주신 것처럼
deployment/~ 부분을 pod/~~으로 바꿔주시면 될거에요. 제가 만든 예제에서는 deployment/{deployment이름}을 이용해서 deployment를 통해 생성한 파드 로그를 보도록 작성했습니다.
@sc303030 은 하나의 파드를 지정해서 로그를 보고자 하신 것이니 pod/{pod이름}을 사용하라고 명령어 실행 결과가 아래처럼 나온 것입니다.

PS C:\Users\gh\fastcampus-kubernetes-main\chapter01> kubectl logs deployment/nginx-deployment -c nginx -f
Found 4 pods, **using pod/nginx-deployment-64779c7579-9xmvr**

다시 실행해보시고 안되시면 또 문의주세요.
